mitodl/micromasters
# Add JWT authentication to open-discussions

#### What are the relevant tickets?
Part of #3479 

#### What's this PR do?
Adds support for syncing user profiles and initializing the session JWT cookie for OD.

#### How should this be manually tested?
Add the following to `.env`:
```
OPEN_DISCUSSIONS_JWT_SECRET=terribly_unsafe_default_jwt_secret_key
OPEN_DISCUSSIONS_BASE_URL=http://172.28.0.1:8063/  # set to the gateway IP for your open-discussions app
OPEN_DISCUSSIONS_API_USERNAME=mitodl
OPEN_DISCUSSIONS_REDIRECT_URL=http://localhost:8063/
OPEN_DISCUSSIONS_COOKIE_DOMAIN=localhost
FEATURE_OPEN_DISCUSSIONS_USER_SYNC=True
```

To test, create a new user under micromasters, fill out your profile, and then go to the url `/discussions/` under the micromasters app. You should be redirected to open discussions and be logged in and can post and comment on channels.
